### DISCUSSION TITLE
I am not familiar with c language, kafka output-plugin when producing messages, how is the message timestamp set? Is it the time the log was read or the time the message was created?

### DISCUSSION THREAD

**Discussion by 1123183721** [fluent/fluent-bit#8890] [Q&A] [answered]
I am not familiar with c language, kafka output-plugin when producing messages, how is the message timestamp set? Is it the time the log was read or the time the message was created?

**Comment by patrick-stephens** [comment, marked as answer]
It's whatever you tell it to parse (not all inputs are log files remember) so either you parse it from the input or it defaults to the current time when the input record is created.

https://docs.fluentbit.io/manual/concepts/key-concepts#timestamp does say this at the bottom too - there's no need to understand the code just check the docs.
